### DISCUSSION TITLE
Getting order completed event

### DISCUSSION THREAD

**Discussion by yoavweber** [kernc/backtesting.py#902] [Q&A] [answered]
Hey People,
Is there any callback/way to get notified when an order gets filled? 

My goal is to create a new order based on the filled order's data.

Thanks!

**Comment by kernc** [comment, marked as answer]
There is nothing else besides the API, which should be simple enough to wrap one's head around.

You can track trades like so:
```py
def init(self):
    self.seen_trades = set()

def next(self):
    for trade in self.trades:
        if trade not in self.seen_trades:
            self.seen_trades.add(trade)
            ...  # Create new order based on filled order's data
    ...
